docs/admin/authentication.md

```diff
@@ -683,3 +683,144 @@ rules:
   verbs: ["impersonate"]
   resourceNames: ["view", "development"]
 ```
+
+## client-go credential plugins
+
+{% assign for_k8s_version="v1.10" %}{% include feature-state-alpha.md %}
+
+`k8s.io/client-go` and tools using it such as `kubectl` and `kubelet` are able to execute an
+external command to receive user credentials.
+
+This feature is intended for client side integrations with IdPs, such as cloud IAM services, and
+protocols not natively supported by `k8s.io/client-go`, such as Kerberos, LDAP, and custom OAuth2
+flavors.
```

Comment:
Would this client-go credential plugin work with SAML or OpenID Connect? My guess is no but you should make it clear whether these two federated identity approaches can be supported by the plugin or not since these two approaches are the most popular ones needed by enterprise customers.